src/components/ContentWrapper.jsx

```diff
@@ -1,51 +1,80 @@
-import ContactsListDataLoader from './ContactsList/ContactsListDataLoader'
+import React, { useState } from 'react'
+import { PropTypes } from 'prop-types'
+
+import { useClient } from 'cozy-client'
+import log from 'cozy-logger'
+import { Content } from 'cozy-ui/transpiled/react/Layout'
+
 import Header from './Header'
 import Toolbar from './Toolbar'
-import { Content } from 'cozy-ui/transpiled/react/Layout'
-import { Query } from 'cozy-client'
-
-const ContentWrapper = () => (
-  <Query
-    query={client =>
-      client
-        .all(DOCTYPE_CONTACTS)
-        .include(['accounts'])
-        .where({
-          $or: [
-            {
-              trashed: {
-                $exists: false
-              }
-            },
-            {
-              trashed: false
-            }
-          ],
-          _id: {
-            $gt: null
-          }
-        })
-        .indexFields(['_id'])
-    }
-  >
-    {({ data: contacts, fetchStatus, hasMore, fetchMore }) => {
-      return (
-        <>
-          {contacts.length >= 1 && <Header right={<Toolbar />} />}
-          <Content>
-            <ContactsListDataLoader
-              contacts={contacts}
-              fetchStatus={fetchStatus}
-              hasMore={hasMore}
-              fetchMore={fetchMore}
-            />
-          </Content>
-        </>
-      )
-    }}
-  </Query>
-)
+import ContactsList from '../components/ContactsList/ContactsList.jsx'
+import SpinnerContact from './Components/Spinner'
+import {
+  useFetchContacts,
+  updateAndSortByFamilyNameGivenNameEmailCozyUrl
+} from '../helpers/contentWrapper'
+import {
+  contactsbyFamilyNameGivenNameEmailCozyUrl,
+  contactsWithoutIndexes
+} from '../helpers/queries'
+
+const ContentWrapper = ({ serviceToLaunch, isServiceLaunched }) => {
+  const client = useClient()
+  const [isContactsWithIndexesFinished, contactsWithIndexes] = useFetchContacts(
+    contactsbyFamilyNameGivenNameEmailCozyUrl
+  )
+  const [
+    isContactsWithNoIndexesFinished,
+    contactsWithNoIndexes
+  ] = useFetchContacts(contactsWithoutIndexes)
+
+  const [hasAlreadySortedContacts, setHasAlreadySortedContacts] = useState(
+    false
+  )
+  const [sortedContacts, setSortedContacts] = useState([])
+
+  const fetchesAreFinished =
+    isContactsWithIndexesFinished && isContactsWithNoIndexesFinished
+
+  // start keepIndexFullNameAndDisplayNameUpToDate service
+  // if never launched before
+  if (serviceToLaunch && isServiceLaunched === false) {
+    log(
+      'info',
+      `Executing keepIndexFullNameAndDisplayNameUpToDate service by Contacts app`
+    )
+    client.collection('io.cozy.triggers').launch(serviceToLaunch)
+  }
+
+  if (!fetchesAreFinished) {
+    return <SpinnerContact size="xxlarge" loadingType="fetching_contacts" />
+  }
+
+  if (fetchesAreFinished && !hasAlreadySortedContacts) {
+    const reworkedContacts = isServiceLaunched
+      ? contactsWithIndexes.concat(contactsWithNoIndexes)
```

Comment:
I'm confused about why this works: if the service has run, do we still have `contactsWithNoIndexes`? And if we do, don't they need to be re-sorted?

Reply:
Oui on peut avoir encore des `contactsWithNoIndexes` car même si le service a déjà été lancé une fois, on peut par la suite créer un nouveau contact sans index (via un connecteur par exemple).

Reply:
Je comprend, mais dans ce cas les contacts sans index ne seront pas triés et seront à la fin de la liste non ? Pourquoi ne pas faire le tri côté client dans ce cas la aussi ?

Reply:
Oui c'est ce comportement qui est attendu.
Parce qu'on veut éviter autant que possible de faire du traitement en front. Dans l'autre cas on n'a pas le choix, mais l'autre cas est voué à disparaître quand les contacts auront tous migré et que le service aura été lancé au moins une fois sur toutes les instances. Pour plus de précision il y a les infos sur le paper https://paper.dropbox.com/doc/SEC_0012_CONTACTS_Beta--A6YS4mP2ytOsATqmPovCsJ~dAg-8TAfiAilZTySqvdaYePjQ